main/src/ca/uwaterloo/flix/language/phase/Deriver.scala

```diff
@@ -802,7 +803,7 @@ object Deriver {
         val (_, caze) = cases.head
 
         val outSym = new Symbol.AssocTypeSym(coerceTraitSym, "Out", loc)
-        val outTpe = caze.tpe
+        val outTpe = Type.mkTuplish(caze.tpes, loc)
```

Comment:
Coerce will extract the old type, i.e. unit for empty, tpe for single term cases, and a tuple for n-ary enums